src/main/java/com/salesforce/servicelibs/BackwardsCompatibilityCheckMojo.java

```diff
@@ -189,15 +195,33 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 options = "";
             }
 
-            Path lockFile = Paths.get(protoSourceRoot, "proto.lock");
+            if (options.toUpperCase().contains("--LOCKDIR")) {
```

Comment:
Move this new code section to a private method that returns the `lockDir`.  It's not your fault that `execute()` is getting really hairy, but it's time to start breaking it into methods.